src/Components/OATHeader/OATHeader.tsx

```diff
@@ -76,204 +92,92 @@ const OATHeader: React.FC<IOATHeaderProps> = (props) => {
-                let accumulatedError = '';
-                for (const error of newFilesErrors) {
-                    accumulatedError += `${error} \n `;
-                }
-
+            const result = await parseFilesToModels({
+                files: files,
+                currentModels: oatPageState.currentOntologyModels,
+                localizationKeys: IMPORT_LOC_KEYS,
+                translate: t
+            });
+            if (result.status === 'Success') {
+                oatPageDispatch({
+                    type: OatPageContextActionType.IMPORT_MODELS,
+                    payload: { models: result.models }
+                });
+            } else if (result.status === 'Failed') {
+                // show error
+                const error =
+                    result.errors?.length > 0
+                        ? result.errors[0]
+                        : {
+                              title: t('OAT.Common.unhandledExceptionTitle'),
+                              message: t('OAT.Common.unhandledexceptionMessage')
+                          };
                 oatPageDispatch({
                     type: OatPageContextActionType.SET_OAT_ERROR,
                     payload: {
-                        title: t('OATHeader.errorFormatNoSupported'),
-                        message: accumulatedError
+                        title: error.title,
+                        message: error.message
                     }
                 });
             }
-            handleFileListChanged(newFiles);
             // Reset value of input element so that it can be reused with the same file
             uploadFolderInputRef.current.value = null;
             uploadFileInputRef.current.value = null;
-            logDebugConsole('debug', '[IMPORT] [END] Files upload.');
+            logDebugConsole('debug', '[IMPORT] [END] Files upload.', result);
         },
-        [
-            oatPageDispatch,
-            oatPageState.currentOntologyModels,
-            t,
-            uploadFileInputRef
-        ]
+        [oatPageDispatch, oatPageState.currentOntologyModels, t]
     );
 
+    const onExportClick = useCallback(() => {
+        logDebugConsole(
+            'info',
+            '[START] Export models to file. {models}',
+            oatPageState.currentOntologyModels
+        );
+
+        const zipResult = createZipFileFromModels({
+            models: oatPageState.currentOntologyModels,
+            localizationKeys: EXPORT_LOC_KEYS,
+            translate: t
+        });
+        if (zipResult.status === 'Success') {
+            zipResult.file.generateAsync({ type: 'blob' }).then((content) => {
+                logDebugConsole(
+                    'info',
+                    '[END] Export models to file. {content}',
+                    content
+                );
+                downloadFile(
+                    content,
+                    `${
+                        oatPageState.currentOntologyProjectName || 'ontology'
```

Comment:
Do we need to be concerned with ProjectName containing invalid filename characters here?

Reply:
oh that's an interesting point. What characters would that be? I guess I could stick to the hard coded name to be safe. This was just a nice cosmetic thing but I doubt there is a good regex that would help us to sanitize those.

Reply:
Something like `^[a-zA-Z0-9](?:[ a-zA-Z0-9._-]*[a-zA-Z0-9])?$`  allows upper/lower/numeric to start; allows numbers, spaces, etc, within the string; doesn't allow trailing spaces or periods

Reply:
that seems fair.

Reply:
Added.